SQF-VM/SQF_commands.c

```diff
@@ -45,6 +45,35 @@ int64_t system_time_ms(void)
 	return (int64_t)(((timebuffer.time * 1000) + timebuffer.millitm));
 #endif
 }
+
+/**
+ * Checks whether the given array is a 3D vector.
+ * If not an error will be thrown
+ */
+char checkVector3(const PVM vm, const PARRAY array)
+{
+	// size check
+	if (array->top != 3)
+	{
+		vm->error(vm, ERR_LEFT ERR_ARRAY ERR_ARRAY_SIZE_EXACT(3), vm->stack);
+
+		return 0;
+	}
+
+	// content type check
+	for (int i = 0; i < array->top; i++)
+	{
+		if (array->data[i]->type != SCALAR_TYPE())
+		{
+			vm->error(vm, ERR_CAN_ONLY_PROCESS ERR_SCALAR " " ERR_ARRAYS, vm->stack);
```

Comment:
please alter error message according to this: https://github.com/X39/sqf-vm/blob/master/SQF-VM/SQF_commands.c#L4028 (also update the `setPos`, `setVelocity` commands to use this helper method)